model/error/event.go

```diff
@@ -113,20 +133,17 @@ type Log struct {
 	Stacktrace   m.Stacktrace
 }
 
-func DecodeEvent(input interface{}, cfg m.Config, err error) (transform.Transformable, error) {
-	if err != nil {
-		return nil, err
-	}
-	if input == nil {
-		return nil, errMissingInput
-	}
-
+func (d decoder) Decode(input interface{}, requestTime time.Time, metadata metadata.Metadata) (m.Transformable, error) {
 	raw, ok := input.(map[string]interface{})
 	if !ok {
 		return nil, errInvalidType
 	}
```

Comment:
Would it be reasonable to instead always assume the input type is an object, and do the type assertion on the outside and change the `Decode` signature to accept `map[string]interface{}`? Then we could get rid of a little bit of boilerplate.

Alternatively, maybe we could make this a naked type assertion straight after a successful call to Validate, as it will have internally checked the type. i.e.

```go
if err := validation.Validate(input, cachedModelSchema); err != nil {
    return nil, err
}
raw := input.(map[string]interface{}) // safe because Validate already checked the type
```

Reply:
Sure, happy to go with a naked type assertion!